includes/class-wc-product-variable.php

```diff
@@ -67,8 +67,10 @@ public function check_stock_status() {
 		} elseif ( $this->backorders_allowed() || $this->get_stock_quantity() > get_option( 'woocommerce_notify_no_stock_amount' ) ) {
 			$set_child_stock_status = 'instock';
 		}
+		
+		$set_child_stock_status = apply_filters( 'woocommerce_product_check_stock_status', $set_child_stock_status, $this );
```

Comment:
Have a separate hook for variable products?

$set_child_stock_status = apply_filters( 'woocommerce_**variable**_check_stock_status', $set_child_stock_status, $this );